rest/src/main/java/org/jboss/pnc/rest/provider/ProductMilestoneProvider.java

```diff
@@ -137,30 +135,52 @@ public void update(Integer id, ProductMilestoneRest restEntity) throws Validatio
     }
 
     public void update(Integer id, ProductMilestoneRest restEntity, String accessToken) throws ValidationException {
-        ProductMilestone milestoneInDb = repository.queryById(id);
-
-        // If we're about to close a milestone (by setting the end date), don't make the closed milestone current
-        if (milestoneInDb.getEndDate() == null && restEntity.getEndDate() != null) {
-
-            ProductVersion productVersion = milestoneInDb.getProductVersion();
 
-            if (productVersion.getCurrentProductMilestone() != null &&
-                    productVersion.getCurrentProductMilestone().getId().equals(milestoneInDb.getId())) {
+        ProductMilestone milestoneInDb = repository.queryById(id);
 
-                productVersion.setCurrentProductMilestone(null);
-                productVersionRepository.save(productVersion);
-            }
+        // we can't modify milestone if it's already released
+        if (milestoneInDb.getEndDate() != null) {
+            log.error("Milestone is already closed: no more modifications allowed");
```

Comment:
The log level error doesn't look appropriate to me. If the user tries to close already closed milestone, it's not a PNC issue. Info level should be fine here.